shell/edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -864,7 +864,7 @@ export default {
   },
 
   created() {
-    this.registerBeforeHook(this.saveMachinePools, 'save-machine-pools');
+    this.registerBeforeHook(this.saveMachinePools, 'save-machine-pools', 1);
```

Comment:
question: why the `1`?

Reply:
the third arg is the priority of the hook and dictates the order they run in. 

```
registerBeforeHook(boundFn, name, priority = 99, boundFnContext) {

...

const hooks = sortBy(this[key] || [], ['priority', 'name']);
```

This should ensure the saveMachinePools hook runs first (to avoid anything else changing if the user cancels). Before they ran in the same order they do now. Adding the priority just makes it clearer saveMchinePools runs first and avoids regressions if someone adds another hook